### ISSUE TITLE
npm run dev loses --headless CLI argument

### ISSUE THREAD

**Issue by bahmutov** [cypress-io/cypress#6151]
Cypress v3.8.2 in Mac

I am trying to test something related to headless browser for https://github.com/cypress-io/cypress/pull/6133 with CLI arguments

```
$ DEBUG=cypress:* npm run dev -- --run-project ~/git/cypress-example-recipes/examples/fundamentals__chrome-remote-debugging --headless --browser chrome
```

The CLI captures `--headless` argument

```
  cypress:server:args options { _: [ '/Users/gleb/.nvm/versions/node/v12.13.0/bin/node', '/Users/gleb/git/cypress/scripts/start.js' ], runProject: '/Users/gleb/git/cypress-example-recipes/examples/fundamentals__chrome-remote-debugging', headless: true, browser: 'chrome', invokedFromCli: false, cwd: '/Users/gleb/git/cypress/packages/server', config: {} } +1ms
```

```
cypress:server:cypress starting in mode run +17ms
  cypress:server:cypress starting Electron +1ms
  cypress:server:cypress electron open arguments [ '--browser=chrome', '--config={}', '--cwd=/Users/gleb/git/cypress/packages/server', '--runProject=/Users/gleb/git/cypress-example-recipes/examples/fundamentals__chrome-remote-debugging' ] +127ms
```
Notice the CLI option `--headless` is not passed and Chrome is visible when running

## Expected behavior

Run Chrome in headless mode when starting with `--headless` CLI option

**Comment by bahmutov** [comment]
turns out, in packages/server/lib/util/args.js we have a whitelist of CLI arguments to pass to Electron. There is `headed` item, thus if you want to pass `--headless` you should pass `--headed false`

```
npm run dev -- --run-project ... --headed false --browser chrome
```

Which is very unintuitive.

**Comment by jennifer-shehane** [comment]
Released in [3.8.3](https://on.cypress.io/changelog#3-8-3)
